File: Top Interview 150/Two Pointers/15_3Sum.py
class Solution(object):
    def threeSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        list1 = set()
        nums.sort()
        for k in range(len(nums)):
            i, j = k+1, len(nums) - 1
            while i < j:
                sum = nums[i] + nums[j] + nums[k]
                if sum == 0:
                    list1.add((nums[k], nums[i], nums[j]))
                    i += 1
                    j -= 1
                elif sum < 0:
                    i += 1
                else:
                    j -= 1
        return list(list1)

File: Top Interview 150/Two Pointers/15_3Sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        set<vector<int>> list1;
        sort(nums.begin(), nums.end());
        for (int k=0; k<nums.size(); k++) {
            int i = k+1;
            int j = nums.size() - 1;
            while (i < j) {
                int sum = nums[i] + nums[j] + nums[k];
                if (sum == 0) {
                    list1.insert({nums[k], nums[i], nums[j]});
                    i += 1;
                    j -= 1;
                }
                else if (sum < 0) {
                    i += 1;
                }
                else {
                    j -= 1;
                }
            }
        }
        return vector<vector<int>>(list1.begin(), list1.end());
    }
};